Automattic/jetpack
# Adds full connection cycle capability to the Connection package.

#### Changes proposed in this Pull Request:
[x] Adds the register method to the Connection package.
[x] Hooks the register verification method for registration verification.
[x] Adds multisite registration capability.
[x] Makes sure the methods are moved with only the needed dependencies.
[x] Makes sure the example plugin is able to connect using only the dependent packages and minimal boilerplate code.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This PR aims to make the Connection package usable without the Jetpack plugin itself.

#### Testing instructions:
* Use [the example client plugin](https://github.com/Automattic/client-example) on a clean WordPress site. It's bare bones, it has one administration page where it displays some debugging information, as well as lets you press a button to run the registration process. You should see that the site properly generates secrets, but can't proceed with user registration because handlers are not yet in place.

#### Proposed changelog entry for your changes:

* N/A
